rust/connlib/tunnel/src/gateway.rs

```diff
@@ -75,13 +64,22 @@ impl GatewayTunnel {
 
     pub fn allow_access(
         &mut self,
-        resource: ResourceDescription<ResolvedResourceDescriptionDns>,
         client: ClientId,
-        expires_at: Option<DateTime<Utc>>,
+        ipv4: Ipv4Addr,
+        ipv6: Ipv6Addr,
```

Comment:
Hm I may be missing something, but where are you pulling these out of the `client` in the message sent by the portal?

```
client: client
```

AFAIK with the above, you'll need to access `ipv4` and `ipv6` like `client.ipv4` etc.

Reply:
Those are already extracted in the eventloop

Reply:
[here](https://github.com/firezone/firezone/pull/6403/files#diff-242d43c4a7a93d0fbd3f84f8af923dbd1ff6d40137ead6834277d654141d60a6R319-R320)

Reply:
Stuff like this ensured by the compiler unless I would be making up IP addresses 😁